### PR TITLE
Align PMUI alternative package label with dotnet.exe

### PR DESCRIPTION
**Pull request by xavierdecoster**
Instead of showing `packageId version x.y.z (or higher)`, the details pane for deprecation alternate package will show `packageId >= x.y.z` instead, which is consistent with the upcoming `dotnet list package` updates for deprecation info, as well as how dependency info is shown in the PMUI.

When the altnerate package is an exact version, the label will display as `packageId`.

### CODE REVIEW

**Review by zivkan — APPROVED** [review]

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs [hunk 1]
```diff
@@ -401,14 +401,18 @@ private string GetPackageDeprecationAlternatePackageText(AlternatePackageMetadat
...
             }
 
             if (!alternatePackageMetadata.Range.HasUpperBound)
             {
-                text += Resources.Label_OrHigher;
+                text = string.Format(
+                    CultureInfo.CurrentCulture,
+                    Resources.Label_DeprecatedPackageIdAndMinVersion,
+                    alternatePackageMetadata.PackageId,
+                    alternatePackageMetadata.Range.MinVersion.ToNormalizedString());
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs]: How about VersionRange.ToLockFileDependencyGroupString? 

That does the whole >= on it's own. 

It also would handle all the ranges correctly if needed.

Comment (xavierdecoster) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs]: That API doesn't exist on `VersionRange` type. It exists on `LibraryRange`.

Comment (xavierdecoster) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs]: Much better though: https://github.com/NuGet/NuGet.Client/pull/2939/commits/3568133c71859c0cae5e8e6cdc3a5c12b0ee9a1e

Introduced `VersionRange.ToPackageVersionRangeSpecificationString(string packageId)` and forwarded the calls.

File: src/NuGet.Core/NuGet.Versioning/VersionRange.cs [hunk 2]
```diff
@@ -402,5 +403,59 @@ public override int GetHashCode()
...
+        /// Format the version range as a package version range specification string.
+        /// </summary>
+        /// <param name="packageId">The package id to be used in the dependency group.</param>
+        /// <example>
+        /// Package.Id >= 1.0.0
+        /// </example>
+        /// <remarks>
+        /// Used to label dependency groups or alternate packages.
+        /// </remarks>
+        public string ToPackageVersionRangeSpecificationString(string packageId)
```
Comment (zivkan) [inline src/NuGet.Core/NuGet.Versioning/VersionRange.cs]: Could you add some tests please.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs]: How does PrettyPrint compare to your change?

I'd imagine it's almost identical :) 
Save for the `()`

File: src/NuGet.Core/NuGet.Versioning/VersionRange.cs (hunk 2 above)
Comment (xavierdecoster) [reply src/NuGet.Core/NuGet.Versioning/VersionRange.cs]: This is existing code so I assumed we had tests. Will check.

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs (hunk 1 above)
Comment (nkolev92) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs]: On another note, should that string be added to the VersionRangeFormatter like the rest of the string representations of the Version Range.

Comment (xavierdecoster) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs]: Hmm... `PrettyPrint` might actually do the trick IMO. See below.

image

File: src/NuGet.Core/NuGet.Versioning/VersionRange.cs (hunk 2 above)
Comment (xavierdecoster) [reply src/NuGet.Core/NuGet.Versioning/VersionRange.cs]: Reverted to use `PrettyPrint` instead :) https://github.com/NuGet/NuGet.Client/pull/2939/commits/03838cdb06c60782d11e46ed60e7266125924609

File: src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs (hunk 1 above)
Comment (xavierdecoster) [reply src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs]: Reverted to use `PrettyPrint` instead :) https://github.com/NuGet/NuGet.Client/pull/2939/commits/03838cdb06c60782d11e46ed60e7266125924609

File: src/NuGet.Core/NuGet.Versioning/VersionRangeFormatter.cs [hunk 3]
```diff
@@ -11,8 +11,6 @@ namespace NuGet.Versioning
     /// </summary>
     public class VersionRangeFormatter : IFormatProvider, ICustomFormatter
     {
-        private const string LessThanOrEqualTo = "<=";
-        private const string GreaterThanOrEqualTo = ">=";
```
Comment (xavierdecoster) [inline src/NuGet.Core/NuGet.Versioning/VersionRangeFormatter.cs:15]: These private fields were unused.

**Review by nkolev92 — APPROVED** [review]
I assume we don't care about the extra `()`? 

I don't have a preference, just want to make sure we're aware of the impact :)

**Comment by anangaur** [comment]
Without the parentheses is fine too. Rather better :)

**Comment by xavierdecoster** [comment]
@nkolev92 is there a similar API to `PrettyPrint` without the parenthesis? Want to avoid yet another string format just to avoid these parenthesis...

**Comment by nkolev92** [comment]
Unfortunately no. 
It's ok if we add it I guess.
There's already a use case (with the assets file stuff :) )

**Comment by xavierdecoster** [comment]
> There's already a use case (with the assets file stuff :) )

Any pointers to code where this already happens?

**Comment by nkolev92** [comment]
I'm talking about the earlier use case of: LibraryRange.ToLockFileDependencyGroupString.

**Comment by xavierdecoster** [comment]
@nkolev92 [this commit](https://github.com/NuGet/NuGet.Client/pull/2939/commits/d05affada2d579c2447add5967bcdda11f99266c) should fix just that: `PrettyPrint` without parentheses. Whilst at it, also made `VersionRangeFormatter` API a little more consistent to `VersionFormatter`.

Once that commit is reviewed, this can be merged.

File: src/NuGet.Core/NuGet.Versioning/VersionRange.cs [hunk 4]
```diff
@@ -103,15 +103,15 @@ public override string ToString()
         /// </summary>
         public virtual string ToNormalizedString()
         {
-            return ToString("N", new VersionRangeFormatter());
+            return ToString("N", VersionRangeFormatter.Instance);
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Versioning/VersionRange.cs:106]: 👏 

The useless objects this whole thing created. :)

**Review by nkolev92 — APPROVED** [review]
Love it. 
👏

**Pull request merged**